source/common/router/router.cc

```diff
@@ -76,6 +76,20 @@ bool convertRequestHeadersForInternalRedirect(Http::HeaderMap& downstream_header
     return false;
   }
 
+  if (!filter_state.hasData<UInt32Accessor>(NumPreviousInternalRedirectFilterStateName)) {
```

Comment:
Comment block?
// Make sure that performing the redirect won't result in exceeding the configured number of redirects allowed for this route.
?

Reply:
Done.